src/uwtools/drivers/forecast.py

```diff
@@ -162,12 +162,28 @@ def job_card(self):
         Turns the resources config object into a batch card for the configured Task.
         """
 
-    def run(self):
-        """
-        Runs the forecast executable with the namelist file and staged input files.
-
-        This will both build the executable and run it.
-        """
+    def run(
+        self,
+        exec_name: str,
+        export: Optional[str] = None,
+        mpi: Optional[str] = None,
+        node_arg: Optional[str] = None,
+        pe_member01: Optional[int] = None,
+        num_of_procs: Optional[int] = None,
+        oversubscribe: Optional[str] = None,
+        procs_per_nodes: Optional[str] = None,
+        ppn: Optional[int] = None,
+        cpu_bind: Optional[str] = None,
+        core: Optional[str] = None,
+        depth: Optional[str] = None,
+        omp_num_threads: Optional[int] = None,
+    ) -> str:
+        """
+        Builds the forecast executable with the user provided run commands.
+
+        This will build the executable.
```

Comment:
I think that "build an executable" will be thought by many developers to mean "compile source code into an executable program" and will be surprising to see in this context. I think something like `Constructs a command to be used to run the forecast executable` might describe this better.